jsonify_stored/models/jsonify_stored_mixin.py

```diff
@@ -0,0 +1,179 @@
+    jsonified_display = fields.Text(
+        compute="_compute_jsonified_display",
+    )
+    jsonified_data = Serialized(compute="_compute_jsonified_data", store=True)
+
+    def _compute_ir_export_id(self):
+        for record in self:
+            record.ir_export_id = record._jsonify_get_exporter()
+
+    def _jsonify_get_exporter(self):
+        """Hook to get the exporter to use."""
+        return False
+
+    def _compute_jsonified_display(self):
+        for record in self:
+            record.jsonified_display = json.dumps(
+                record.jsonified_data, sort_keys=True, indent=4
+            )
+
+    def _compute_jsonified_data(self):
+        """Compute JSON data for current recordset.
+
+        Optimized to always group by lang and parser
+        even when large mixed recordsets are updated.
+        """
+        default_lang = self._get_default_lang()
+        for grouper, group in self._compute_jsonified_data_groupby():
+            records = self.browse([r.id for r in group])
+            exporter = grouper.exporter_id
+            if not exporter:
+                records.jsonified_data = False
+                _logger.error("%s have no exporter", str(records._name))
+                continue
+            lang_code = grouper.lang_code or default_lang
+            # `get_json_parser` is cached
+            parser = exporter.get_json_parser()
+            self._update_jsonified_data(records.with_context(lang=lang_code), parser)
+
+    def _compute_jsonified_data_groupby(self):
+        _grouper_and_sorter = self._jsonify_data_grouper_and_sorter
+        return groupby(self.sorted(_grouper_and_sorter), key=_grouper_and_sorter)
+
+    def _get_default_lang(self):
+        return get_lang(self.env)
+
+    @staticmethod
+    def _jsonify_data_grouper_and_sorter(rec):
+        lang_code = None
+        if "lang_id" in rec._fields:
+            # Support translated records.
+            lang_code = rec.lang_id.code
+        return GrouperAndSorter(rec.ir_export_id, lang_code)
+
+    def _update_jsonified_data(self, records, parser):
+        json_vals = records._get_jsonified_data(parser)
+        for record, data in zip(records, json_vals):
+            record.jsonified_data = data
+
+    def _get_jsonified_data(self, parser):
+        # Hook to override data
+        return self.jsonify(parser)
+
+    @api.model
+    def cron_update_json_data_for(
+        self,
+        model_name,
+        chunk_size=5,
+        domain=None,
+        langs=None,
+        job_params=None,
+    ):
+        """Generate jobs to compute JSON data for given model.
+
+        If the curent model has a `lang_id` field, one job-ification per lang will be triggered.
+        """
+        _logger.info("cron_update_json_data_for: %s", model_name)
+        model = self.env[model_name]
+        domain = domain or []
+        if "lang_id" in model._fields and not langs:
+            langs = model._json_data_compute_get_all_langs()
```

Comment:
Seems a bit unnecessary having to maintain `_json_data_compute_get_all_langs` with a raw sql that only works if the field is stored, and having to worry about cases with inherits.

Why not simply get the list of active res.lang records, instead?
Most databases have very few languages installed, so the performance imp you get with the distinct select sql query is not much.

Reply:
I don't get your point. What shall I do w/ the list of active languages? Here I want to load only the records for a given language.

Reply:
I mean, `_json_data_compute_get_all_langs` could be removed and simply do this:

```suggestion
            langs = self.env["res.lang"].search([]).mapped("code")
```

In most databases you'll find not more than 3~4 languages installed, so no need to get the list of languages from the records table IMO.

Reply:
well, yeah but not all languages might be used for a given model, that's why I'm looking only for real records' values.